src/sys/unix.rs

```diff
@@ -542,6 +543,36 @@ impl Socket {
         unsafe { self.setsockopt(libc::IPPROTO_IP, libc::IP_TTL, ttl as c_int) }
     }
 
+    #[cfg(target_vendor = "apple")]
+    pub fn set_bound_interface(&self, name: &str) -> io::Result<()> {
+        const IP_BOUND_IF: libc::c_int = 25;
+        let name = CString::new(name)?;
+        unsafe {
+            let index = libc::if_nametoindex(name.as_ptr());
```

Comment:
I think we need to check whether `index` is valid. If the index is 0, that indicates an error and we should inspect `errno` in that case.

Reply:
Hm ideally this'd just be one "syscall", so perhaps the argument here should be the index and not the name?

Reply:
good input, I’ll split it out into two functions instead

Reply:
Updated to remove the additional name to interface index lookup since it's already available in libc. I believe this resolves your comments